frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1167,8 +1167,17 @@ end
 --]]
 
 function ReaderHighlight:exportToDocument(page, item)
+    local setting = G_reader_settings:readSetting("save_document")
+    if setting == "disable" then return end
     logger.dbg("export highlight to document", item)
-    self.ui.document:saveHighlight(page, item)
+    local can_write = self.ui.document:saveHighlight(page, item)
+    if can_write == false and not self.warned_once then
+        self.warned_once = true
+        UIManager:show(InfoMessage:new{
+            text = _("Document is read only. Cannot write to it"),
```

Comment:
May be something a bit more explicite, like "Document is read only. Your highlights will not be written into it." There's surely a better wording :) @Frenzie?

Reply:
Probably best to make it two lines. Something like this:
```
This document is set to read-only.
Highlights can't be saved into a read-only document.
```

Reply:
Well, I'm not good at english and no good at verbosity

```
Cannot write highlights: document is read only.
```
 ?

Or maybe with explicit implications:
```
Cannot write highlights: document is read only.
Hightlights won't be saved on this document.
If you want to edit the document itself please move it to a writable dir first.
```

Reply:
Users may not even know highlights get written in the PDF itself.
And we want to assure them the thing they just highlighted will be highlighted next time, as it is saved in the metadata.lua.
So, may be something a bit less negative, and verbose:

> The highlights you'll be making on this document will be saved in the setting file, but won't be written in the PDF itself, as this PDF file is in a read-only location.
If you wish your highlights to be saved in the PDF file itself, just move it to a writable directory, so it can be updated.

Reply:
@poire-z the first statement is true, the second is not in this particular case.

metadata is saved to history/ because sidecar dir can't be created in RO. I don't know if metadata is ported when a user copies/moves documents within the program, but when doing it outside the user is expected to replace the file manually in order to keep highlights in place. Hence my "copy/move first", because that's actually easier to explain that "copy/move, open once, exit and copy metadata from history to sidecar".

I'm actually happy avoiding the crash. Any nits I could have on the string are just from user pov when facing with this message (probably android, reading a document from sdcard) and to avoid future bug reports related to this issue.

Reply:
So, just add a "first" :)
> If you wish your highlights to be saved in the PDF file itself, move this PDF file _first_ to a writable directory, so it can be rewritten.

and let the discovery of the technical steps involved to the user.
(I guess most will not care if we tell them they are saved anyway in the first sentence.)

Reply:
updated the PR to match your suggestion. I've skipped the PDF references by calling them "documents" but I'm aware that this should apply to pdf's only.